inc/media.php

```diff
@@ -1692,21 +1692,29 @@ function media_nstree($ns)
     $ns_parts = explode(':', $ns);
     $tmp_ns = '';
     $pos = 0;
+    $insert = false;
     foreach ($ns_parts as $level => $part) {
-        if ($tmp_ns) $tmp_ns .= ':' . $part;
-        else $tmp_ns = $part;
+        if ($tmp_ns) {
+            $tmp_ns .= ':' . $part;
+        } else {
+            $tmp_ns = $part;
+        }
 
-        // find the namespace parts or insert them
-        while ($data[$pos]['id'] != $tmp_ns) {
+        // find the namespace parts
+        while (array_key_exists($pos, $data) && $data[$pos]['id'] != $tmp_ns) {
             if (
                 $pos >= count($data) ||
                 ($data[$pos]['level'] <= $level + 1 && Sort::strcmp($data[$pos]['id'], $tmp_ns) > 0)
             ) {
-                array_splice($data, $pos, 0, [['level' => $level + 1, 'id' => $tmp_ns, 'open' => 'true']]);
+                $insert = true;
```

Comment:
Unless I'm missing something, I believe this is not necessary because the foreach processes the elements of the "current namespace" (e.g. `aaa:bbb:ccc` -> `aaa`, `bbb` and `ccc`), and the while loop checks for each element if it's present in the media directory's structure. 

Therefore, if we start inserting an entry in $data at any point in that second loop because it was not found (say, `aaa`, then by definition nothing exists below that newly created tree branch, so we will keep inserting until we reach the end of the foreach loop (`bbb`and then `ccc`).

Reply:
Ok, if that is what this code is doing, your explanation makes sense.

PS. Sorry about commenting in the wrong repository. I was reacting to a GitHub notification about a change and failed to check.